be2-scala/src/main/scala/ch/epfl/pop/model/objects/Transaction.scala

```diff
@@ -27,10 +27,30 @@ final case class TxIn(
 )
 
 final case class Transaction(
-  Version: Int,
-  TxIn: List[TxIn],
-  TxOut: List[TxOut],
-  )
+  version: Int,
+  inputs: List[TxIn],
+  outputs: List[TxOut],
+  lockTime: Int,
+) {
+  lazy val transactionId = {
+    val strings = collection.mutable.ListBuffer.empty[String]
```

Comment:
Interesting use of buffer, wouldn't have thought of using them :) ! I don't know what's the typical size of `inputs`, but remember that the list buffer will only be more efficient than a list if the collection is "large"